internal/server/repository_workerauth.go

```diff
@@ -520,38 +520,25 @@ func (r *WorkerAuthRepositoryStorage) findCertBundles(ctx context.Context, worke
 	return certBundle, nil
 }
 
-
-// VerifyAuthorizableWorkerKeyIds returns all the worker key identifiers that
+// FilterToAuthorizedWorkerKeyIds returns all the worker key identifiers that
 // are authorizable from the slice of key identifiers provided to the function.
-func (r *WorkerAuthRepositoryStorage) VerifyAuthorizableWorkerKeyIds(ctx context.Context, workerKeyIds []string) ([]string, error) {
-	const authorizedWorkerQuery = `
-	select distinct w.worker_key_identifier 
-	from 
-	    worker_auth_certificate_bundle as w
-	where
-	    w.worker_key_identifier in (?)
-`
-	const op = "server.(WorkerAuthRepositoryStorage).VerifyAuthorizableWorkerKeyIds"
+func (r *WorkerAuthRepositoryStorage) FilterToAuthorizedWorkerKeyIds(ctx context.Context, workerKeyIds []string) ([]string, error) {
+	const op = "server.(WorkerAuthRepositoryStorage).FilterToAuthorizedWorkerKeyIds"
 	if len(workerKeyIds) == 0 {
 		return nil, nil
 	}
-	rows, err := r.reader.Query(ctx, authorizedWorkerQuery, []interface{}{workerKeyIds})
-	if err != nil {
+	var bundles []*WorkerCertBundle
+	if err := r.reader.SearchWhere(ctx, &bundles, "worker_key_identifier in (?)", []interface{}{workerKeyIds}, db.WithLimit(-1)); err != nil {
 		return nil, errors.Wrap(ctx, err, op)
 	}
```

Comment:
I took "custom query" to mean crafting a query that gets passed directly to [Query](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/db/read_writer.go#L236-L251) or [Exec](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/db/read_writer.go#L221-L234) instead of calling things like [SearchWhere](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/db/read_writer.go#L497-L517) that construct a query. There are many examples that fall into this category, generally they are in a `query.go` file in the domain packages:

- [session/query.go](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/session/query.go#L8-L425)
- [iam/query.go](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/iam/query.go#L6-L111)
- [target/query.go](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/target/query.go#L3-L58)
- [server/query.go](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/server/query.go#L3-L33)
- [credential/vault/query.go](https://github.com/hashicorp/boundary/blob/51de0878c37658203f64c8080f04815ba4b317f9/internal/credential/vault/query.go#L3-L189)